examples/plot_metric_examples.py

```diff
@@ -0,0 +1,474 @@
+# Metric Learning for Kernel Regression
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# An algorithm very similar to NCA but for regression is Metric
+# Learning for Kernel Regression (MLKR). It will optimize for the average
+# leave-one-out *regression* performance from a soft-nearest neighbors
+# regression.
+#
+# - See more in the :ref:`User Guide <mlkr>`
+# - See more in the documentation of the class :py:class:`MLKR
+#   <metric_learn.mlkr.MLKR>`
+#
+# To illustrate MLKR, let's use the dataset
+# `sklearn.datasets.make_regression` the same way as we did with the
+# classification  before. The dataset will contain: 100 points of 5 features
+# each, among which 3 are random noise but informative (used to generate the
+# regression target from a linear model, and two are random noise with the
+# same magnitude
+
+X_reg, Y_reg = make_regression(n_samples=100, n_informative=3, n_features=5,
+                               shuffle=True)
+
+######################################################################
+# Let's plot the dataset as is
+
+plot_tsne(X_reg, Y_reg, plt.cm.Oranges)
+
+######################################################################
+# And let's plot the dataset after transformation by MLKR:
+mlkr = metric_learn.MLKR()
+X_mlkr = mlkr.fit_transform(X_reg, Y_reg)
+plot_tsne(X_mlkr, Y_reg, plt.cm.Oranges)
+
+######################################################################
+# Points that have the same value to regress are now closer to each
+# other ! This could improve the performance of
+# `sklearn.neighbors.KNeighborsRegressor` for instance.
+
+
+######################################################################
+# Constraints
+# ^^^^^^^^^^^
+#
+# To learn the right metric, so far we have always given the labels of the
+# data to supervise the algorithms. However, in many applications,
+# it is easier to obtain information about whether two samples are
+# similar or dissimilar. For instance, when annotating a dataset of face
+# images, it is easier for an annotator to tell if two faces belong to the same
+# person or not, rather than finding the ID of the face among a huge database
+# of every person's faces.
+#
+# Fortunately, one of the strength of metric learning is the ability to
+# deal with such data. Indeed, some of the algorithms we've mentioned have
+# alternate ways to pass some supervision about the metric we want to learn.
+# The way to go is to pass a 3D array `pairs` of pairs, as well as an array
+# of labels `y` such that for each `i` between `0` and `n_pairs`
+# we want `X[i, 0, :]` and `X[i, 1, :]` to be similar if `y[i] == 1`, and we
+# want them to be dissimilar if `y[i] == -1`. In other words, we want to
+# enforce a metric that projects similar points closer together and
+# dissimilar points further away from each other.
+# (See also the section: :ref:`weakly_supervised_section`.)
+#
+# This kind of input is possible for ITML, SDML, and MMC.
```

Comment:
put this sentence before the pointer to the doc

Reply:
done